src/components/cta/SubscribeCTA.js

```diff
@@ -1,72 +1,146 @@
-import { React, useRef } from 'react'
+import { React, useRef, useState } from 'react';
+
+import { useFormik } from 'formik';
+import * as yup from 'yup';
 
 import Button from '@mui/material/Button';
 import Box from '@mui/material/Box';
 import TextField from '@mui/material/TextField';
 import Typography from '@mui/material/Typography';
 import { useTheme } from '@mui/material/styles';
 
-export default function SubscribeCTA({text='Join thousands of others for our newsletter'}) {
+
+import Backdrop from '@mui/material/Backdrop';
+import Modal from '@mui/material/Modal';
+import Fade from '@mui/material/Fade';
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+  fontFamily: "Mulish",
+};
+
+function TransitionsModal({title, message, open, setOpen}) {
+   const handleOpen = () => setOpen(true);
+   const handleClose = () => setOpen(false);
+
+  return (
+    <div>
+
+      <Modal
+        aria-labelledby="transition-modal-title"
+        aria-describedby="transition-modal-description"
+        open={open}
+        onClose={handleClose}
+        closeAfterTransition
+        BackdropComponent={Backdrop}
+        BackdropProps={{
+          timeout: 500,
+        }}
+      >
+        <Fade in={open}>
+          <Box sx={style}>
+            <Typography id="transition-modal-title" variant="h6" component="h2">
+              Thank you
+            </Typography>
+            <Typography id="transition-modal-description" sx={{ mt: 2 }}>
+              Have a great day!
+            </Typography>
+          </Box>
+        </Fade>
+      </Modal>
+    </div>
+  );
+}
+
+export default function SubscribeCTA({
+  text = 'Join thousands of others for our newsletter',
+}) {
   const theme = useTheme();
-  const emailInput = useRef(null);
-  const validateEmail = (email) => /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email)
-  
-  const onSubmit = () => {
-    let email = emailInput.current.children[1].children[0].value
-    if(validateEmail(email)){
-      alert('good submit to capture '+email)
-    }
-    alert('Bad Email: '+email)
-  }
-  
+    const [open, setOpen] = useState(false);
+
+
+  const validationSchema = yup.object({
+    email: yup
+      .string('Enter your email')
+      .email('Enter a valid email')
+      .required('Email is required'),
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values) => {
+      // alert(JSON.stringify(values, null, 2));
+
+    const email = JSON.stringify(values)
```

Comment:
@ardeay im a little stuck on how to return the value(email) and make it accessible to the POST request?